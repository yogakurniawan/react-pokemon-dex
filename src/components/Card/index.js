import React from 'react';
import PropTypes from 'prop-types';
import FlatButton from 'material-ui/FlatButton';
import { Card, CardTitle, CardActions, CardMedia } from 'material-ui/Card';
import { extractId, capitalizeFirstLetter, getPokemonImage } from 'utils';

const styles = {
  media: {
    height: 150,
    width: 150,
    margin: '0 auto'
  },
  button: {
    fontSize: 14,
    margin: '0 auto'
  }
};

function CardComponent({ pokemon, classes, showDetail }) {
  const { url, name } = pokemon;
  const pokemonId = extractId(url);
  const pokemonImage = getPokemonImage(pokemonId);
  return (
    <Card style={{ marginBottom: 20 }}>
      <CardMedia>
        <img src={pokemonImage} alt={name} />
      </CardMedia>
      <CardTitle titleStyle={{ fontSize: 16 }} title={capitalizeFirstLetter(name)} />
      <CardActions>
        <FlatButton
          primary
          label="Show Detail"
          style={styles.button}
          onClick={() => showDetail(pokemonId)}
        />
      </CardActions>
    </Card>
  );
}

CardComponent.propTypes = {
  pokemon: PropTypes.object.isRequired,
  showDetail: PropTypes.func.isRequired
};

CardComponent.defaultProps = {
};

export default CardComponent;
