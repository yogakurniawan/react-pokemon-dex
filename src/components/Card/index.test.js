import React from 'react';
import { mount } from 'enzyme';
import injectTapEventPlugin from 'react-tap-event-plugin';
import renderer from 'react-test-renderer';
import { Card } from 'material-ui/Card';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import CardComponent from './index';

injectTapEventPlugin();

const pokemon = {
  name: 'bulbasaur'
}

const Wrapper = mount(
  <MuiThemeProvider>
    <CardComponent pokemon={pokemon} showDetail={jest.fn()} />
  </MuiThemeProvider>
);

describe('Card', () => {

  it('Should render self and subcomponents', () => {
    expect(Wrapper.length).toEqual(1);
  });

  it('Should contain <Card />', () => {
    expect(Wrapper.find(Card).length).toEqual(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <CardComponent pokemon={pokemon} showDetail={jest.fn()} />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});