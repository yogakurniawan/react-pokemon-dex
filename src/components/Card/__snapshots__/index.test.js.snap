// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "marginBottom": 20,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <img
          alt="bulbasaur"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          }
        />
      </div>
      
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": 16,
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }
      >
        Bulbasaur
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#00bcd4",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 14,
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": "0 auto",
            "marginRight": 8,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Show Detail
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
